### PR TITLE
*: refine some code and comments according to golangci-lint

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve? 

I use golangci-lint 1.41.1 to run:

```shell
golangci-lint run
```

According to the results, I refined some code and comments, such as:

* Inefficient assignment
* Code complexity
* Misspelling
* Deadcode

### What is changed and how it works?

See above.

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

 - No release note

### CODE REVIEW

File: pkg/lightning/restore/meta_manager.go [hunk 1]
```diff
@@ -535,6 +535,7 @@ func (m *dbTaskMetaMgr) CheckAndPausePdSchedulers(ctx context.Context) (pdutil.U
 	pauseCtx, cancel := context.WithCancel(ctx)
 	conn, err := m.session.Conn(ctx)
 	if err != nil {
+		cancel()
```
Comment (glorv) [inline pkg/lightning/restore/meta_manager.go:538]: perfer call `defer cancel()` afer L535

File: pkg/restore/client.go [hunk 2]
```diff
@@ -181,7 +181,7 @@ func (rc *Client) InitBackupMeta(c context.Context, backupMeta *backuppb.BackupM
 
 		var ddlJobs []*model.Job
 		// ddls is the bytes of json.Marshal
-		ddls, err := reader.ReadDDLs(c)
+		ddls, _ := reader.ReadDDLs(c)
```
Comment (glorv) [inline pkg/restore/client.go:184]: shall we check insted of ingore of ignore this error? @3pointer

**Review by glorv — COMMENTED** [review]
rest LGTM

**Comment by tisonkun** [comment]
@JmPotato where does this change set come from? It seems strange a PR with title "make some changes".

**Comment by JmPotato** [comment]
> @JmPotato where does this change set come from? It seems strange a PR with title "make some changes".

Updated the description. PTAL.

File: pkg/lightning/restore/meta_manager.go (hunk 1 above)
Comment (JmPotato) [reply pkg/lightning/restore/meta_manager.go:538]: This function returns the cancel function to the caller. So I think `defer` should not be used here.

https://github.com/pingcap/br/blob/df27ac2045fa5f3942bae79a0aa20b58689145bf/pkg/lightning/restore/meta_manager.go#L639-L643

**Review by nolouch — COMMENTED** [review]
lgtm about the lint.

File: pkg/restore/client.go (hunk 2 above)
Comment (overvenus) [reply pkg/restore/client.go:184]: We should check the error.

**Review by overvenus — COMMENTED** [review]
Rest LGTM

**Comment by JmPotato** [comment]
@overvenus @glorv PTAL, thx!

**Review by overvenus — APPROVED** [review]
LGTM, thanka!

**Review by glorv — APPROVED** [review]

**Pull request merged**